Marketplace cards branch

zekai128 opened the pull request:
Add marketplace cards to marketplace. Next step is to populate the marketing cards with actual card data fetched from the database.

zekai128 committed:
add cd workflow

zekai128 committed:
fix pyflake errors

zekai128 committed:
change the name of test.py file to not conflict with test folder

matticyang reviewed (commented):
we split views.py into 2 other python files, this is commit is not up to date

zekai128 commented:
> we split views.py into 2 other python files, this is commit is not up to date

OK. I added that when rebasing. I'll remove views.py in another commit.

zekai128 committed:
remove views.py:

larryworm1127 commented on client/src/components/Marketplace/index.tsx:
```diff
@@ -1,10 +1,50 @@
   return (
     <Box sx={{ display: 'flex' }}>
-
+        <Grid container spacing={5}>
```
In react, you can do
```
{list.map(item => (
  <Grid item xs={12} sm={6} md={4} lg={3} xl={2.4}>
      <MarketplaceTaskCard title='T1' description='Card for T1' height = '300' />
  </Grid>
)}
```
That way you can just put the data in the list which in this case is the title, description and height, and don't have to repeat the JSX syntax 20 times

zekai128 replied:
OK. I was looking for something like this.

larryworm1127 replied:
There's also
```
{condition && (
  <JSXElement/>
)}
```
which will only display `<JSXElement/>` if the condition is true, and also
```
{condition 
  ? <JSXElement1/>
  : <JSXElement2/>
}
```
which behaves exactly the same as normal ternary expression

zekai128 replied:
Not needed here, but maybe can be used if we want to do some extra filtering after the API fetch from database?

zekai128 committed:
remove test file:

matticyang reviewed (commented):
some line addition can be removed to for further pr

larryworm1127 commented on client/src/components/Shared/MarketplaceTaskCard.tsx:
```diff
@@ -0,0 +1,43 @@
+import * as React from 'react'
```
Wait, isn't this just the same as `MarketingTaskCard` component in Shared folder? I'm guessing you created a new file just to add the `height` prop? If that's the case, then simply adding a `height?` into the props to the existing component should be enough.

zekai128 replied:
good point. I added a height here to try and do the masonry setup for the marketplace tab, but since we aren't doing different heights for now should just use the same Card component

zekai128 committed:
trim whitespace at end of line for style check

larryworm1127 commented on client/src/components/Shared/MarketplaceTaskCard.tsx:
```diff
@@ -0,0 +1,43 @@
+    height: height
+    };
+  return (
+    <Card style = {cardStyle} sx={{ width: '100%' }}>
```
Don't use `style` for material UI component. Use their `sx` prop which is very useful when you want your entire app to have the same standard for margins, paddins, etc. The rest of the style also works the same for `sx` prop.

zekai128 replied:
removed style in latest commit, now using sx for the height param.

zekai128 committed:
remove MarketplaceTaskCard and replace with shared MarketingTaskcard component
